activesupport/lib/active_support/message_verifier.rb

```diff
@@ -198,8 +203,35 @@ def decode(data)
       end
 
       def generate_digest(data)
-        require "openssl" unless defined?(OpenSSL)
-        OpenSSL::HMAC.hexdigest(OpenSSL::Digest.const_get(@digest).new, @secret, data)
+        OpenSSL::HMAC.hexdigest(@digest, @secret, data)
```

Comment:
This will raise an exception with a nicer error message - "Unsupported digest algorithm (X)" instead of "wrong constant name X" - but will break things for people creating instances of `MessageVerifier` with a symbol as the `digest` kwarg. This was exactly the case of the test I had to fix. All documentation shows `digest` being a string, so IMHO, this is OK. We can always call change the initializer to `@digest = digest&.to_s || "SHA1"` and make everyone happy. Looking for feedback.

I actually thought about changing the initializer to `@digest = OpenSSL::Digest.new(digest || "SHA1")`, which would cause instance initialization to fail if an unsupported digested is provided, instead of failing once a method on the instance is called. I'm a fan of the "fail fast" principle, but I understand that it's a more significant departure.